src/mon/Monitor.cc

```diff
@@ -2271,6 +2271,120 @@ void Monitor::do_health_to_clog(bool force)
   health_status_cache.summary = summary;
 }
 
+namespace {
+  enum class scrubbed_or_deepscrubbed_t { SCRUBBED, DEEPSCRUBBED };
+
+  void print_unscrubbed_detailed(const std::pair<const pg_t,pg_stat_t> pg_entry,
+				 bool *section_opened,
+				 stringstream &ss,
+				 Formatter *f,
+				 scrubbed_or_deepscrubbed_t how_scrubbed) {
+
+    const auto& pg_stat(pg_entry.second);
+    if (!*section_opened) {
+      if (f != nullptr) {
+	f->open_array_section("unscrubbed_pgs");
+      }
+
+      *section_opened = true;
+    }
+
+    if (f != nullptr) {
+      f->open_object_section("unscrubbed_pg");
+      f->dump_stream("pgid") << pg_entry.first;
+      if (how_scrubbed == scrubbed_or_deepscrubbed_t::SCRUBBED) {
+	f->dump_stream("last_scrub_stamp") << pg_stat.last_scrub_stamp;
+      } else if (how_scrubbed == scrubbed_or_deepscrubbed_t::DEEPSCRUBBED) {
+	f->dump_stream("last_deep_scrub_stamp")
+	  << pg_stat.last_deep_scrub_stamp;
+      }
+      f->close_section();
+    } else {
+      ss << "pg " << pg_entry.first << " is not ";
+      if (how_scrubbed == scrubbed_or_deepscrubbed_t::SCRUBBED) {
+	ss << "scrubbed, last_scrub_stamp "
+	   << pg_stat.last_scrub_stamp;
+      } else if (how_scrubbed == scrubbed_or_deepscrubbed_t::DEEPSCRUBBED) {
+	ss << "deep-scrubbed, last_deep_scrub_stamp "
+	   << pg_stat.last_deep_scrub_stamp;
+      }
+      ss << std::endl;
+    }
+  }
+
+  void print_unscrubbed_count(int pgs_count,
+			      bool *section_opened,
+			      stringstream &ss,
+			      Formatter *f) {
+    if (f != nullptr) {
+      *section_opened = true;
+      f->open_object_section("unscrubbed_pgs");
+      f->dump_stream("number_unscrubbed_pgs") << pgs_count;
+    } else {
+      ss << "number unscrubbed pgs: " << pgs_count;
+    }
+  }
```

Comment:
drop this helper.. there is only one caller and the arguments and *section_opened part is needlessly complex